### DISCUSSION TITLE
What are the pros and cons of Vercel compared to a server?

### DISCUSSION THREAD

**Discussion by jlorezz** [vercel/community#59] [Help] [answered]
What are the pros and cons of Vercel compared to a server?

**Comment by leerob** [comment, marked as answer]
### Pros

- You don't have to manage servers. There's no version patching, server updates, or expensive hardware needed. Instead, you get to take advantage of [modern infrastructure](https://vercel.com/features/infrastructure).
- You can scale _infinitely_, whether it's your personal site or [large companies like Carhartt, Github, IBM, McDonald’s, Uber](https://vercel.com/customers), etc.
- You get the best parts of multi-cloud. For example, supporting [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions) as well as [Edge Functions](https://vercel.com/edge).

### Cons

- It's a mental model shift. For example, you can't write to the file system in serverless. Certain problems require different solutions in serverless architectures. [More here.](https://vercel.com/docs/concepts/solutions/overview)
- [Database connections need to be handled differently.](https://twitter.com/leeerob/status/1430177168307412998)
- You have _too much free time_ to work on your product instead of wrangling servers. Okay, this one was a joke 😄
